### ISSUE TITLE
client socket transport - select unused port/ port 0

### ISSUE THREAD

**Issue by gco-bmx** [microsoft/vscode-languageserver-node#665]
In the client when using a socket transport, to support having multiple instances of a language server running, when multiple windows are open, I want to select an unused port.

I believe I can specify a port of 0 and the OS will select an unused port, 
I've very new to javascript/typescript but 
there doesn't appear to be any path in the code where the actually connected to port number will then be passed to the 
--socket={port} option which tells the server which port to connect to 

Is my understanding correct? or am I missing something?

**Comment by dbaeumer** [comment]
No, your assumption is correct. PR welcome to add this functionality.
